server/handlers/validation.go

```diff
@@ -402,7 +412,7 @@ func validateRoot(gun string, oldRoot, newRoot []byte, store storage.MetaStore)
 // checkRoot errors if an invalid rotation has taken place, if the
 // threshold number of signatures is invalid, if there are an invalid
 // number of roles and keys, or if the timestamp keys are invalid
-func checkRoot(oldRoot, newRoot *data.SignedRoot, timestampKey data.PublicKey) error {
+func checkRoot(oldRoot, newRoot *data.SignedRoot, gun string, store storage.MetaStore) error {
```

Comment:
Actually, sorry, I just realized that after the rebase, a lot of this code can go.  Will refactor it out.